<launch>
  <arg name="namespace" default="/camera" />
  <arg name="name" default="kinfu" />
  <arg name="response_topic" default="/$(arg name)_response_topic" />
  <arg name="request_topic" default="/$(arg name)_request_topic" />

  <node pkg="kinfu" type="kinfu" name="$(arg name)" output="screen">
    <param name="prefix_topic" value="$(arg namespace)" />
    <param name="camera_info_topic" value="$(arg namespace)/depth/camera_info" />
    <param name="depth_image_topic" value="$(arg namespace)/depth/image_raw" />
    <param name="image_topic" value="$(arg namespace)/rgb/image_raw" />

    <param name="autostart" value="true" />
    <param name="response_topic" value="$(arg response_topic)" />
    <param name="request_topic" value="$(arg request_topic)" />
    <param name="extract_known_points" value="false" />

    <param name="depth_height" type="int" value="480" />
    <param name="depth_width" type="int" value="640" />
  </node>
</launch>

